src/services/blockchain/ethereum/ethereum-blockchain-service.ts

```diff
@@ -65,16 +71,20 @@ async function attempt<T>(
  * @param txData - Transaction to write.
  * @param walletBalance - Available funds.
  */
-function handleInsufficientFundsError(txData: TransactionRequest, walletBalance: BigNumber): void {
-  const txCost = (txData.gasLimit as BigNumber).mul(txData.maxFeePerGas!)
-  if (txCost.gt(walletBalance)) {
+function handleInsufficientFundsError(txData: TransactionRequest, walletBalance: bigint): void {
+  const gasLimit = Utils.coalesce(() => txData.gasLimit, 'No gas limit')
```

Comment:
And here simpler `if (!gasLimit) throw new Error()` seems more appropriate than coalesce.

Reply:
Yeah, I see the point.